server/pbenchinacan/deploy

```diff
@@ -1,26 +1,84 @@
+#
+# Other inputs which can be provided by environment variables include:
+#   - the name and pull policy for the Pbench Server container image
+#   - the location of the Pbench Dashboard app files and the Nginx Favicon
+#   - the location of the Pbench Git checkout (used to locate the above)
+#
+# In all cases, suitable defaults are provided for any value which is undefined.
+#
 
-# Update the favicon
-podman unshare rm /srv/pbench/public_html/favicon.ico
-podman unshare cp color-square.ico /srv/pbench/public_html/favicon.ico
+#+
+# Configuration
+#-
 
-IMAGE=images.paas.redhat.com/pbench/pbench-server:main
+# Pbench Server container image
+PB_SERVER_IMAGE=${PB_SERVER_IMAGE:-"images.paas.redhat.com/pbench/pbench-server:main"}
+PB_SERVER_IMAGE_PULL_POLICY=${PB_SERVER_IMAGE_PULL_POLICY:-"newer"}
 
-# First ensure /srv/pbench is owned by the "pbench" user INSIDE the container we run.
-podman run --rm --volume /srv/pbench:/srv/pbench:Z ${IMAGE} chown -R pbench:pbench /srv/pbench
+# Locations on the host
+PB_DASHBOARD_DIR=${PB_DASHBOARD_DIR:-${GITTOP:=$(git rev-parse --show-toplevel)}/dashboard/build}
+NGINX_FAVICON=${NGINX_FAVICON:-${GITTOP:=$(git rev-parse --show-toplevel)}/server/lib/config/color-square.ico}
+PB_DEPLOY_FILES=${PB_DEPLOY_FILES:-${HOME}/Deploy}
+SRV_PBENCH=${SRV_PBENCH:-/srv/pbench}
 
-# Deploy the actual Pbench Server instance.
+#+
+# Deployment
+#-
+
+# Update the dashboard code, removing any existing code and copying in fresh.
+# We expect that the host dashboard directory already exists and contains a
+# build of the Pbench Dashboard application; however, if it is missing, we
+# deploy the Pbench Server without a Dashboard deployment.
+podman unshare rm -r ${SRV_PBENCH}/public_html/dashboard
+if [[ -d "${PB_DASHBOARD_DIR}" ]]; then
+    podman unshare cp -r ${PB_DASHBOARD_DIR} ${SRV_PBENCH}/public_html/dashboard
+else
+    echo "Warning:  deploying the Pbench Server without a Dashboard" >&2
+    podman unshare mkdir -m 0755 -p ${SRV_PBENCH}/public_html/dashboard
+fi
+
+# Update the Nginx favicon and provide some static fallback pages
+podman unshare rm ${SRV_PBENCH}/public_html/favicon.ico
+podman unshare cp ${NGINX_FAVICON} ${SRV_PBENCH}/public_html/favicon.ico
+podman unshare cp /usr/share/nginx/html/404.html /usr/share/nginx/html/50x.html \
+    ${SRV_PBENCH}/public_html/
+
+# Ensure ${SRV_PBENCH} is owned by the "pbench" user INSIDE the container.
+podman run --rm --volume ${SRV_PBENCH}:/srv/pbench:Z ${PB_SERVER_IMAGE} \
+    chown -R pbench:pbench /srv/pbench
+
+# Run the Pbench Server.
 podman run \
-    --name pbench-server \
-    --rm \
     --detach \
+    --name pbench-server \
     --network host \
-    --volume ${HOME}/Deploy/etc/rsyslog.conf:/etc/rsyslog.conf:Z \
-    --volume ${HOME}/Deploy/etc/rsyslog.d:/etc/rsyslog.d:Z \
-    --volume ${HOME}/Deploy/pbench-server.cfg:/opt/pbench-server/lib/config/pbench-server.cfg:Z \
-    --volume /srv/pbench:/srv/pbench:Z \
-    ${IMAGE} /sbin/init
+    --pids-limit -1 \
```

Comment:
@portante, @ndokos, @npalaska:  the CI deployment was specifying a pids limit of _zero_, which makes no obvious sense -- I believe that that would block all `fork()`'s, which seems like a challenging situation, especially since the Pbench Server is not the only pid in the container.  The doc says that the max is `2048` and that `-1` means no limit...are we happy with no-limit, or would we like to run with the default, which is what the Staging deployment was using previously.

Reply:
Default, thanks.

Reply:
Done; and applied to the dependency services, as well.